config/routes.rb

```diff
@@ -13,20 +13,38 @@
   end
 
   namespace :editor do
-    root 'welcome#index'
+    get "/(:controller_name)/(:id)/(:action)", to: redirect {|params, request|
+      base_url = "#{request.scheme}://#{request.host_with_port}/sites/#{request.host}/editor"
+      path = [params[:controller_name], params[:id], params[:action]].compact.join("/")
+      if path.empty?
+        base_url
+      else
+        "#{base_url}/#{path}"
+      end
+    }
+  end
+
+  namespace :sites do
+    root "welcome#index"
+
+    resources :sites, only: %i(), path: "/", param: :fqdn, fqdn: %r{[^/]+} do
```

Comment:
It is easy to memorize FQDN for human.
(I cannot memorize site ID, It's too difficult for me)